From 46c6d08f03867375252fdb1278d7d35dca921954 Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Mon, 1 Sep 2014 09:26:17 -0400
Subject: [md] raid0: check for bitmap compatability when changing raid levels

Message-id: <1409563584-9000-76-git-send-email-Jes.Sorensen@redhat.com>
Patchwork-id: 89975
O-Subject: [RHEL7.1 PATCH 75/82] md/raid0: check for bitmap compatability when changing raid levels.
Bugzilla: 1085530 1126348
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: NeilBrown <neilb@suse.de>

If an array has a bitmap, then it cannot be converted to raid0.

Reported-by: Xiao Ni <xni@redhat.com>
Signed-off-by: NeilBrown <neilb@suse.de>
(cherry picked from commit a8461a61c241a25afedbe493c13d98a6e0cf4246)
Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/raid0.c b/drivers/md/raid0.c
index c4d420b..b81666e 100644
--- a/drivers/md/raid0.c
+++ b/drivers/md/raid0.c
@@ -704,6 +704,12 @@ static void *raid0_takeover(struct mddev *mddev)
   *  raid10 - assuming we have all necessary active disks
   *  raid1 - with (N -1) mirror drives faulty
   */
+
+ if (mddev->bitmap) {
+  printk(KERN_ERR "md/raid0: %s: cannot takeover array with bitmap\n",
+         mdname(mddev));
+  return ERR_PTR(-EBUSY);
+ }
  if (mddev->level == 4)
   return raid0_takeover_raid45(mddev);
 
-- 
1.7.1